Demo/HYPForms/HYPSampleCollectionViewController.m

```diff
@@ -29,13 +29,19 @@ @implementation HYPSampleCollectionViewController
 - (instancetype)initWithDictionary:(NSDictionary *)dictionary
 {
     HYPFormsLayout *layout = [[HYPFormsLayout alloc] init];
-
     self = [super initWithCollectionViewLayout:layout];
     if (!self) return nil;
 
-    layout.dataSource = self;
+    self.layout = layout;
     self.initialValues = dictionary;
 
+    [self.collectionView registerClass:[HYPImageFormFieldCell class]
+            forCellWithReuseIdentifier:HYPImageFormFieldCellIdentifier];
+
+    self.collectionView.dataSource = self.dataSource;
+
+    NSLog(@"Initializated with %ld sections", (long)[self.collectionView numberOfSections]);
```

Comment:
It's weird, but initializing the controller in a unit test doesn't trigger any of the dataSource methods:

```objc
- (NSInteger)numberOfSectionsInCollectionView:(UICollectionView *)collectionView;
- (NSInteger)collectionView:(UICollectionView *)collectionView numberOfItemsInSection:(NSInteger)section;
```

So the next time these are called is when a change has been made to it's data, so the dataSource no longer contains the correct state.

That's this weird thing is needed.

Maybe would be better to wrap it with a `isTest` call